pkg/operator/ceph/object/livenessprobe.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package object allows a ceph config file to be stored in Kubernetes and mounted as volumes into
+// Ceph daemon containers.
+package object
+
+import (
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookv1 "github.com/rook/rook/pkg/apis/rook.io/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+// configureLivenessProbe returns the desired liveness probe for a given daemon
+func configureLivenessProbe(daemon rookv1.KeyType, container v1.Container, healthCheck cephv1.BucketHealthCheckSpec) v1.Container {
```

Comment:
Instead of creating this method, seems like we could factor `config.ConfigureLivenessProbe()` to work for the object store as well. Maybe that method just needs to be factored into two methods, one that takes a map (for multiple liveness probes in the spec), and another method that takes a single livenessprobe object.

Reply:
IMO, with the latest changes, we should keep it separate.

Reply:
Which particular latest changes?

Reply:
In `configureLivenessProbe()` we are passing 2 arguments and for `config.ConfigureLivenessProbe()` we are passing 3 arguments.

Reply:
Ok, and why can't we re-use `config.ConfigureLivenessProbe()`?

Reply:
We can re-use `config.ConfigureLivenessProbe()` but we will have to pass an extra argument which is not required in this case [configureLivenessProbe()](https://github.com/rook/rook/pull/5836/commits/23f0ac7f9c7157383bc860b259c1964df46454c1#diff-785678985f266006c1d386ca1f7cf55fR178) but if that needed we can still do it.

Reply:
It's better to avoid code duplication so let's re-use `config.ConfigureLivenessProbe()`.

Reply:
sounds good

Reply:
forgot to mention that if re-use [`config.ConfigureLivenessProbe()`](https://github.com/rook/rook/blob/master/pkg/operator/ceph/config/livenessprobe.go#L30) or not any way we'll have to write a separate method for  `cephobjectstore` crd [like here](https://github.com/rook/rook/pull/5836/commits/23f0ac7f9c7157383bc860b259c1964df46454c1#diff-785678985f266006c1d386ca1f7cf55fR179).

Reply:
I don't see why you cannot handle the rgw config in `ConfigureLivenessProbe()` directly. You can write a `cephv1.GetRgwLivenessProbe` and just add it to `probeFnMap`

Reply:
so, we have discussed the reason behind doing this way. Thanks for approving.